include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -403,9 +415,33 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
     }
   }
 
+  if(spatial_discretization == SpatialDiscretization::HDIV)
+  {
+    AssertThrow(
+      use_divergence_penalty == false || use_continuity_penalty == false,
+      dealii::ExcMessage(
+        "Penalty terms can not be used in case of Hdiv function spaces. Think about using L2 instead, or deactivate the penatly terms."));
+    if(temporal_discretization != TemporalDiscretization::BDFCoupledSolution)
+      AssertThrow(
+        preconditioner_velocity_block != MomentumPreconditioner::InverseMassMatrix &&
+          preconditioner_velocity_block != MomentumPreconditioner::BlockJacobi &&
+          preconditioner_velocity_block != MomentumPreconditioner::Multigrid,
+        dealii::ExcMessage(
+          "Multigrid, BlockJacobi, and InverseMassMatrix as preconditioner are not availiable for HDIV. Use either PointJacobi or None instead, or change to L2 discretization. "));
```

Comment:
Here, I would also not refer to L2. I think this type of commentes is hard to maintain and does not really provide additional information. Typically, one does not switch to a different discretization because a preconditioner is not available for Hdiv.

Reply:
Is it maybe better to check for the ones that work?